src/recorder.ts

```diff
@@ -13,7 +13,16 @@ import { FunctionInfo } from "./registry";
 import commonPathPrefix from "./util/commonPathPrefix";
 import { getTime } from "./util/getTime";
 
-const processRecordingShouldAlwaysBeActive = "APPMAP_RECORDER_PROCESS_ALWAYS" in process.env;
+const kAppmapRecorderProcessAlwaysEnvar = "APPMAP_RECORDER_PROCESS_ALWAYS";
+const processRecordingShouldAlwaysBeActive =
+  kAppmapRecorderProcessAlwaysEnvar in process.env &&
```

Comment:
I don't think you need this check, `isTruthy()` is going to correctly return `false` on `undefined`.